File: Biocomputing/Week8 Scripting REGEX/Lecture15_ORF.R
#Regex ORF Challenge in R
#10/11/17, MMD

#Load package
library('stringr')

#Read in fasta file, each line becomes an item in a vector
InFile=scan("R.mendax.1.fasta",what=character(),sep="\n")

#Loop over vector of lines
for (Line in 1:length(InFile)){
  #Operate only on lines that do not include >, skips header lines
  #Note that str_detect returns a logical T/F
  if (!str_detect(InFile[Line],">")){
    #Find the ORF in each sequence line
    match = str_extract(InFile[Line],"ATG([ATCG]{3})+(TAA|TAG|TGA)")
    #Print the ORF to standard out
    print(match)
  }
}

File: Biocomputing/Week8 Scripting REGEX/Lecture15_ORF.py
#Regex ORF Challenge in Python
#10/11/17, MMD

#Load module
import re

#Open fasta file to read
InFile=open("R.mendax.1.fasta","r")

#Loop through open file
for Line in InFile:
    #Remove end of line character
    Line = Line.strip()
    #Operate only on lines that do not include >, skips header lines
    if ">" not in Line:
        #Find the ORF in each sequence line
        match = re.search(r"ATG([ATCG]{3})+(TAA|TAG|TGA)",Line)
        #Print the ORF to standard out, note that group(0) is the full match
        print(match.group(0))

InFile.close()

